### PR TITLE
Document for engineer onboarding timeline & expectations

### PR DESCRIPTION
**Pull request by jwilander**
>This is the general timeline and expectations for new engineers starting at Mattermost. It is meant to guide new engineers, help them focus on the right tasks, and set expectations to help them be successful at Mattermost. It is not a hard timeline and circumstances may shorten, lengthen or re-order the timeline.

This has already been reviewed by Corey, most of the dev leads and some other devs.

### CODE REVIEW

**Comment by jwilander** [comment]
FYI @coreyhulen

File: source/process/engineer-expectations.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+##### Week 1: Focus on environment setup and introductions
+- Set up equipment and environment (e.g. complete the day 1 checklist)
+  - Get laptop, set up developer environment
+  - Clone enterprise repo
+  - Log in to all accounts (Gmail, GitHub, JIRA, OneLogin, Community, Community Daily)
+  - Expense equipment
+  - HR set up (benefits, health insurance, etc.)
+- Meet the team
+  - Meet with a different team member for 30min each day
+- Meet with your mentor (dev lead)
```
Comment (cpanato) [inline source/process/engineer-expectations.md]: The new hired will always have the dev lead as a mentor? Or we will give a change to others assume this position and also grown :)

Comment (jwilander) [reply source/process/engineer-expectations.md]: This is only for "mentor for the week", so the first week will always be the dev lead then we'll rotate through the next few weeks with different mentors

Comment (jwilander) [reply source/process/engineer-expectations.md]: Updated it to be more clear

**Pull request merged**